src/v/serde/parquet/shredder.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "serde/parquet/schema.h"
+#include "serde/parquet/value.h"
+
+namespace serde::parquet {
+
+// A shredded value that has an assigned repetition level
+// (for determining list index) and definition level (for
+// determining where the null value in the hierarchy is).
+struct shredded_value {
+    int32_t schema_element_index;
+    value val;
+    uint8_t rep_level;
```

Comment:
Yeah I've been assuming `int8_t` but for definition depth, it might be possible to hit 255. I think we limit most of our schemas to max depth of 64 or so, but there can be blowup of nesting using logical types of maps and lists... Probably safer to go to `int16_t`, we don't need to save the bytes really either. I will make this change together with the encoding in another followup PR.